src/scriptune.js

```diff
@@ -136,6 +89,41 @@ export function setMasterVolume(value) {
  */
 export async function play(sheet, options = {}) {
     const tracks = parseSheet(sheet);
+    const currentTime = audioContext.currentTime;
+    const oscillators = [];
+    const promises = [];
+    const onAbort = () => oscillators.forEach(oscillator => oscillator.stop());
+
+    options.signal?.addEventListener('abort', onAbort);
+
+    Object.values(tracks).forEach(notes => {
```

Comment:
This is too heavy for the runtime. When more than 3k notes are scheduled, execution is aborted. Although 3k notes is technically enough (`6 minutes of sixteenth notes at 120 BPM), this should be addressed. Unblocking the main thread with an immediate timeout isn’t enough, because it's the number of oscillators that break it.

A better solution might be to schedule only the next 10 seconds of notes, then pause for 9 seconds before scheduling the next 10 seconds. That precision is sufficient and won’t disrupt playback when the tab becomes inactive like it did in the previous version.